Sources/OpenAPIRuntime/Multipart/MultipartFramesToRawPartsSequence.swift

```diff
@@ -342,8 +342,13 @@ extension MultipartFramesToRawPartsSequence {
             switch stateMachine.nextFromPartSequence() {
             case .returnNil: return nil
             case .fetchFrame:
+              let frame : Upstream.AsyncIterator.Element?
```

Comment:
Could you break out these async sequence iteration fixes into a separate PR? These I think we'll be able to land soon, but the other ones (CharacterSet and URL), I'd like us to fix in swift-corelibs-foundation instead.

The shape of this fix is correct, but I wonder if we can make the `#if` more selective to the stdlib having `next(isolation:)`, rather than to switch by platform?

Reply:
Let's use `#if compiler(>=6.0)`.

Reply:
Can we specify `Linux` and `compiler(>=6.0)`? There's no issue compiling it in Xcode as is.

Reply:
It's because complete concurrency checking isn't enabled by default, but if you turn it on, you'll see it on Darwin as well. No need to limit this to Linux, this is not a platform-specific issue, it's a Swift version-specific fix.

Reply:
Done. For the record, what's the difference between `compiler(>=6.0)` and `swift(>=6.0)`?